src/main/java/com/rarchives/ripme/App.java

```diff
@@ -39,15 +39,23 @@
     private static final History HISTORY = new History();
 
     public static void main(String[] args) throws MalformedURLException {
-        Utils.configureLogger();
-        System.setProperty("apple.laf.useScreenMenuBar", "true");
-        System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
-        logger = Logger.getLogger(App.class);
-        logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
-
-        if (args.length > 0) {
+        CommandLine cl = getArgs(args);
+        if (args.length > 0 && cl.hasOption('v')) {
+            System.out.println(UpdateUtils.getThisJarVersion());
+            handleArguments(args);
+        } else if (args.length > 0) {
+            Utils.configureLogger();
+            System.setProperty("apple.laf.useScreenMenuBar", "true");
+            System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
+            logger = Logger.getLogger(App.class);
+            logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
             handleArguments(args);
```

Comment:
I think the `if` and `else if` blocks could be combined under a single `if` block that checks for the args length, nesting an `if` that checks for the `v` flag and executes the code specific to the flag, otherwise the rest under `else` block. That'd avoid doing the same check twice when the `v` flag is omitted:

```java
    public static void main(String[] args) throws MalformedURLException {
        CommandLine cl = getArgs(args);
        if (args.length > 0) {
            if (cl.hasOption('v')) {
                System.out.println(UpdateUtils.getThisJarVersion());
                handleArguments(args);
            } else {
                Utils.configureLogger();
                System.setProperty("apple.laf.useScreenMenuBar", "true");
                System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
                logger = Logger.getLogger(App.class);
                logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
                handleArguments(args);
            }
        } else {
            Utils.configureLogger();
            System.setProperty("apple.laf.useScreenMenuBar", "true");
            System.setProperty("com.apple.mrj.application.apple.menu.about.name", "RipMe");
            logger = Logger.getLogger(App.class);
            logger.info("Initialized ripme v" + UpdateUtils.getThisJarVersion());
            MainWindow mw = new MainWindow();
            SwingUtilities.invokeLater(mw);
        }
    }
```